### ISSUE TITLE
calling host function from host/device function in thrust/mr/allocator.h:109

### ISSUE THREAD

**Issue by allisonvacanti** [thrust/thrust#1049]
```
Building CUDA object CMakeFiles/thrust.example.cuda.custom_temporary_allocation.dir/examples/cuda/custom_temporary_allocation.cu.o
src/thrust/thrust/mr/allocator.h(109): warning: 
calling a __host__ function("
thrust::mr::stateless_resource_allocator<int,  ::thrust::system::cuda::detail::cuda_memory_resource<& ::cudaMalloc, & ::cudaFree,  ::thrust::cuda_cub::pointer<void> > > ::stateless_resource_allocator
") 
from a __host__ __device__ function("
thrust::mr::stateless_resource_allocator<int,  ::thrust::system::cuda::detail::cuda_memory_resource<& ::cudaMalloc, & ::cudaFree,  ::thrust::cuda_cub::pointer<void> > > ::stateless_resource_allocator [subobject]
") is not allowed
```

**Comment by allisonvacanti** [comment]
This is being triggered by [`custom_temporary_allocation.cu:160`](https://github.com/thrust/thrust/blob/master/examples/cuda/custom_temporary_allocation.cu#L155-L160):

```
  thrust::host_vector<int> h_input(num_elements);
  // ...
  thrust::cuda::vector<int> d_input = h_input;
```

I've dug around for a while in the vector, allocator, and memory resource code and didn't see anything that would cause this warning.

**Comment by allisonvacanti** [comment]
Closing as a duplicate of #1007.
